app/assets/stylesheets/components/form/_form.scss

```diff
@@ -5,3 +5,18 @@
 .form__input-container--medium {
   max-width: 140px;
 }
+
+%form-row-2up {
```

Comment:
I _personally_ don't see why that's a conflict. It's a specific use case (I generally favour vertical forms for readability) however, as this is an admin interface, I'd propose treating it as not the same as front end, is an acceptable deviation.